setup.py

```diff
@@ -81,6 +81,7 @@ def read(*rnames):
     EXTENSIONS_DICT = {
         "core": ("expression", "numbers", "rules", "pattern"),
         "builtin": ["arithmetic", "numeric", "patterns", "graphics"],
+        "algorithm": ["parts"],
```

Comment:
I am a little concerned about the eagerness to use Cython instead of understanding exactly what's wrong and fixing it. 

As you have noted yourself, premature cython slows down development. 

Therefore independent of this I have in mind that we should add an environment variable to setup which is tested to determine whether we want Cython at all.

Reply:
I put in Cython this just to check that it would be possible. I can remove it now.

Reply:
I realize what I write may come across as a bit complaining. To be clear, this isn't wrong. It is just somewhat annoying to me because it causes endless hours in pain in development for maybe only a few seconds speedup where what we really need is minutes of speedup.

It  goes in a direction that isn't as helpful as it might be intended to be if we were instead focused on the more the bigger performance gains.

Reply:
Done.

Reply:
Thanks for your tolerance, understanding, and patience. Again, I will try to put in a PR so that we can develop in a way that doesn't require cython when it isn't wanted.